components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -7895,13 +7896,18 @@ public API getAPIbyUUID(String uuid, String organization) throws APIManagementEx
                 throw new APIMgtResourceNotFoundException(msg);
             }
         } catch (APIPersistenceException e) {
+            if (e.getMessage().contains("does not exist")) {
+                throw new APIMgtResourceNotFoundException(e);
+            }
             throw new APIManagementException("Failed to get API", e);
         } catch (OASPersistenceException e) {
             throw new APIManagementException("Error while retrieving the OAS definition", e);
         } catch (ParseException e) {
             throw new APIManagementException("Error while parsing the OAS definition", e);
         } catch (AsyncSpecPersistenceException e) {
             throw new APIManagementException("Error while retrieving the Async API definition", e);
+        } catch (IllegalArgumentException e) {
```

Comment:
Do we not send any APIPersistenceException from the persistent layer?.. If we do, do we have a requirement to catch a IllegalArgumentException?

Reply:
Removed via : https://github.com/wso2/carbon-apimgt/pull/11114/commits/e942ee49f9801090a2fc64e0dcd033619fee2eb9